pylitterbot/robot/litterrobot3.py

```diff
@@ -127,7 +127,7 @@ def _parse_sleep_info(self) -> None:
             )
 
         # Handle older API sleep start time
-        if self.sleep_mode_enabled and not start_time:
+        if self.sleep_mode_enabled and not start_time and isinstance(sleep_mode_active, str):
```

Comment:
This does not need to test for instance of `str`. `and sleep_mode_active` should be enough.

My question is what would cause it to be `None` in the first place.

Reply:
patch changed to check for `and sleep_mode_active`

Reply:
> That's precisely what the vendor's API returns -- that device is actually "deleted" from my account (it doesn't show on the vendor's app), and practically every field is `null` in the response.

Can you add a sample response for this and add tests around it? It sounds like there may be more that needs to be captured for "deleted" devices.

Reply:
I'll give it a try -- not too familiar with the Python ecosystem, tbh.

For future reference, this is `resp[0]` at https://github.com/natekspencer/pylitterbot/blob/main/pylitterbot/account.py#L153 that causes this failure:

```[{'litterRobotId': '**REDACTED**', 'litterRobotSerial': None, 'litterRobotNickname': 'Cosmonaut', 'deviceType': 'udp', 'cycleCount': None, 'totalCycleCount': None, 'cycleCapacity': None, 'newCycleCapacity': None, 'savedCycleCapacity': None, 'isDFITriggered': None, 'isDf1Triggered': None, 'isDf2Triggered': None, 'isDfsTriggered': None, 'isManualReset': None, 'savedIsManualReset': None, 'previousDFITriggered': None, 'DFICycleCount': None, 'savedCycleCount': None, 'cleanCycleWaitTimeMinutes': None, 'cyclesAfterDrawerFull': None, 'nightLightActive': None, 'panelLockActive': None, 'sleepModeActive': None, 'sleepModeTime': None, 'powerStatus': None, 'unitStatus': None, 'sleepModeEndTime': None, 'sleepModeStartTime': None, 'lastSeen': None, 'setupDate': None, 'isOnboarded': False, 'didNotifyOffline': None, 'autoOfflineDisabled': True}]
```